packages/fleet/lib/supervisor.ts

```diff
@@ -0,0 +1,278 @@
+import type { DatabaseClient } from './db/client.js';
+import { logger } from './utils/logger.js';
+import * as nodes from './models/nodes.js';
+import * as deployments from './models/deployments.js';
+import { Err, Ok } from '@nangohq/utils';
+import type { Result } from '@nangohq/utils';
+import { FleetError } from './utils/errors.js';
+import type { Deployment, Node } from './types.js';
+import { setTimeout } from 'node:timers/promises';
+import type { NodeProvider } from './node-providers/node_provider.js';
+
+type Action =
```

Comment:
I know Action is an overloaded term in Nango but I could find a better name. Suggestions are welcomed

Reply:
Operation is more accurate I feel, but still very generic

Reply:
good idea. changing for operation